// floes.proto
// protobuf3 definition of the communication scheme between the server and 
// clients

syntax = "proto3";

package floes;

service FloesService {
    rpc GetModel(FloesMessage) returns (FloesMessage) {}
    rpc ContributeModel(FloesMessage) returns (FloesMessage) {}
    rpc Subscribe(FloesMessage) returns (stream FloesMessage) {}
}

message FloesMessage {
    string msg = 1;
    repeated Tensor weights = 2;
    optional string timestamp = 3;
}

message Tensor {
    repeated int32 shape = 1;
    bytes data = 2;
}
